### DISCUSSION TITLE
Should we look for GDS in HSI?

### DISCUSSION THREAD

**Discussion by hughsie** [fwupd/fwupd#6097] [Ideas]
We can read a couple of MSRs to know if new enough microcode is loaded: https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/technical-documentation/gather-data-sampling.html

Is this something that we should do? It's a little far away from "platform integrity" but we could do it pretty easily. @superm1 comments most welcome.

**Comment by superm1** [comment]
I think it's useful because microcode is also distributed with the platform firmware and there is no guarantee that the OS has upgraded it.

That being said this is a fuzzy category. 

How about just call it "mitigations for CVE blah are loaded" and then it can be solved either by platform firmware or OS loaded microcode.

**Reply by hughsie** [reply]
https://github.com/fwupd/fwupd/pull/6106 perhaps?
